### ISSUE TITLE
Ambipolar model with recombination reactions

### ISSUE THREAD

**Issue by aborner1** [sparta/sparta#4]
## Summary

There seems to be an issue when the ambipolar model is used in conjunction with recombination reactions (neutral or ambipolar), as running the code in parallel returns a segmentation fault, and running serially with Valgrind shows memory leaks in fix_ambipolar.cpp and react_bird.cpp.

## Type of Issue

Bug Report

## SPARTA Version (Bug Report)

22Jun18

## Expected Behavior (Bug Report)

Code should be able to handle recombination reactions (neutrals or ions) when ions and electrons are present (ambipolar model).

## Actual Behavior (Bug Report)

Segmentation fault.

## Steps to Reproduce (Bug Report)

Run the _ambi_ example with the attached input script and TCE file (one recombination reaction was added) on 8 processors. Code should return a segmentation fault.

## Files

[in-ambi.txt](https://github.com/sparta/sparta/files/2291309/in-ambi.txt)
[air-tce.txt](https://github.com/sparta/sparta/files/2291310/air-tce.txt)

**Comment by stanmoore1** [comment]
@aborner1 looks like this errors out in the latest version of SPARTA that is in our internal SVN repo. It fails with:

```
ERROR: Ambipolar collision model does not yet support recombination reactions (../collide.cpp:286)
```

Should be updated here in a PR soon.

**Comment by sjplimp** [comment]
@aborner1 @stanmoore1 posted a PR today that treats the error properly, so about
to be merged

**Comment by stanmoore1** [comment]
Fixed by #9, closing #4.
